src/stream.jl

```diff
@@ -691,10 +693,12 @@ function writedata!(output::IO, input::Buffer)
     end
     nwritten::Int = 0
     while buffersize(input) > 0
-        n = Base.unsafe_write(output, bufferptr(input), buffersize(input))
+        n = GC.@preserve input Base.unsafe_write(output, bufferptr(input), buffersize(input))
         consumed!(input, n)
         nwritten += n
+        GC.safepoint()
     end
+    GC.safepoint()
```

Comment:
Why the extra safe point here?

Reply:
Added it to catch looping calls over `writedata!` where nothing is written. May not be strictly necessary, it's been a while since I was looking at this.